microformats.go

```diff
@@ -248,12 +276,18 @@ func (p *parser) walk(node *html.Node) {
 				}
 			case "e":
 				value = new(string)
-				*value = getTextContent(node)
+				*value = strings.TrimSpace(getTextContent(node))
 				var buf bytes.Buffer
+
 				for c := node.FirstChild; c != nil; c = c.NextSibling {
+					p.replaceHref(c)
 					html.Render(&buf, c)
 				}
-				htmlbody = strings.TrimSpace(buf.String())
+
+				htmlbody = strings.Replace(buf.String(), "/>", " />", -1)
+				htmlbody = strings.TrimRightFunc(htmlbody, func(r rune) bool {
+					return r == ' '
```

Comment:
here, the test suite is wrong.  They are including the leading whitespace, but not the trailing whitespace, which doesn't make any sense. Either all of it should be trimmed, or none of it.